packages/gatsby-theme-store/src/sdk/masterData/index.ts

```diff
@@ -0,0 +1,85 @@
+export type DataInput = {
+  [key: string]: unknown
+}
+
+export type MasterDataSaveResponse = {
+  DocumentId: string
+  Href: string
+  Id: string
+}
+
+export type MasterData = (
+  entity: string
+) => {
+  /**
+   * Create a new document
+   */
+  create: (data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * `upsert` query updates a document if it exists, or creates a new.
+   */
+  upsert: (data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * Similar to `upsert`, the `upsertial` query updates partial data of
+   * document if it exists, or creates a new.
+   *
+   * The difference is that data that is not requested for updating will
+   * be kept.
+   */
+  upsertial: (data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * `update` updates an existing document.
+   *
+   * Similar to `upsert`, however it does not create a document does not exist.
+   */
+  update: (id: string, data: DataInput) => Promise<MasterDataSaveResponse>
+  /**
+   * `findUnique` method lets you retrieve a single document by id.
+   */
+  findUnique: (id: string) => Promise<unknown>
+  /**
+   * `delete` method deletes a document by id.
+   */
+  delete: (id: string) => Promise<unknown>
+}
+
+type Method = 'POST' | 'GET' | 'PATCH' | 'DELETE' | 'PUT'
+
+const fetchy = (
+  url: string,
+  method: Method,
+  data?: DataInput
+): Promise<MasterDataSaveResponse> =>
+  new Promise((resolve, reject) => {
+    fetch(url, {
+      headers: {
+        'Content-Type': 'application/json',
+        accept: 'application/vnd.vtex.ds.v10+json',
+      },
+      method,
+      body: JSON.stringify(data),
+    }).then((response) => {
+      if (response.ok) {
+        resolve(response.json())
+      } else {
+        reject(new Error())
+      }
+    })
+  })
+
+const masterDataApi: MasterData = (entity) => {
```

Comment:
why not using a hook ?

Reply:
because it can be used by node and not just react.